src/cart/crt_iv.c

```diff
@@ -2502,17 +2517,32 @@ handle_ivupdate_response(const struct crt_cb_info *cb_info)
 	struct crt_iv_ops	 *iv_ops;
 	int			 rc;
 
+	iv_ops = crt_iv_ops_get(iv_info->uci_ivns_internal,
+				iv_info->uci_class_id);
+	D_ASSERT(iv_ops != NULL);
```

Comment:
Actually daos will hold IV NS refrence before the IV call (iv rpc issue), and only release it once the IV call finished, so this will make sure IV namespace will not be destroyed until all IV operation are finished/exit, so all iv_ops here should not be null in any case. see ds_iv_ns_get/put and ds_iv_ns_stop.

Reply:
Sure the originator will hold the ivns reference, but some random node in the middle that services part of the iv request might be destroying ivns when IV rpc arrives. if it fails to find ivns during the lookup then it should gracefully exit the rpc. If it gets ivns request then it will prevent engine from shutting down until that iv request has been processed and ivns ref is released

Reply:
this is callback for IV update, the ivns should be hold by crt_ivu_update_issue() before set it to callback_info, which should block the IVNS being freed?

Reply:
ah fair point. i thought this was ivupdate handler, not the handler for the response